cmd/kube-proxy/app/server_others.go

```diff
@@ -218,6 +218,8 @@ func NewProxyServer(config *componentconfig.KubeProxyConfiguration, cleanupAndEx
 
 	iptInterface.AddReloadFunc(proxier.Sync)
 
+	metrics.RegisterMetrics()
```

Comment:
Personally I'm not convinced about this change. I agree it's more consistent with the codebase,
but on the other hand, syncProxyRules metric makes sense only in case of ip tables and IPvs. It doesn't make any sense in case of userspace proxy.

Maybe that's not that big deal, but probably someone from networking team should make a final decision.

@freehan @bowei ?

Reply:
kindly ping ~

Reply:
@wojtek-t 

I agree with your opinion - syncProxyRules metric makes sense only in case of iptables and IPVS. So, I changed to register metrics only in iptables and IPVS proxy mode. Please take a look when you have a chance. Thanks!